src/tree/hist/evaluate_splits.h

```diff
@@ -133,20 +134,51 @@ template <typename GradientSumT, typename ExpandEntry> class HistEvaluator {
  public:
   void EvaluateSplits(const common::HistCollection<GradientSumT> &hist,
                       GHistIndexMatrix const &gidx, const RegTree &tree,
-                      std::vector<ExpandEntry>* p_entries) {
+                      std::vector<ExpandEntry>* p_entries,
+                      std::vector<ExpandEntry>* p_entries_sub,
+                      std::vector<ExpandEntry>* p_evaluate_entries,
+                      std::vector<std::vector<std::vector<GradientSumT>>>* p_histograms,
+                      std::vector<std::vector<uint16_t>>* p_local_threads_mapping,
+                      const common::OptPartitionBuilder* p_opt_partition_builder,
+                      RegTree* p_tree,
+                      const bool colsample_enabled) {
+    const std::vector<uint32_t> &cut_ptr = gidx.cut.Ptrs();
+    const size_t n_bins = gidx.cut.Ptrs().back();
+    size_t n_features = gidx.cut.Ptrs().size() - 1;
+    std::vector<std::vector<std::vector<GradientSumT>>>& histograms = *p_histograms;
+    std::vector<std::vector<uint16_t>>& local_threads_mapping = *p_local_threads_mapping;
     auto& entries = *p_entries;
+    auto& entries_sub = *p_entries_sub;
+    auto& evaluate_entries = *p_evaluate_entries;
     // All nodes are on the same level, so we can store the shared ptr.
     std::vector<std::shared_ptr<HostDeviceVector<bst_feature_t>>> features(
         entries.size());
     for (size_t nidx_in_set = 0; nidx_in_set < entries.size(); ++nidx_in_set) {
       auto nidx = entries[nidx_in_set].nid;
       features[nidx_in_set] =
           column_sampler_->GetFeatureSet(tree.GetDepth(nidx));
+          auto features_set = features[nidx_in_set]->ConstHostSpan();
+    }
+    std::vector<std::shared_ptr<HostDeviceVector<bst_feature_t>>> features_sub;
```

Comment:
I'm not entirely sure what you are trying to do here..

Reply:
it's required for merging evaluate and sync kernels